docs/usage.rst

```diff
@@ -117,6 +117,72 @@ there are no ``"conformsTo"`` uris set at all. But they can be explicitly set:
 Note, updating ``"conformsTo"`` does not change what the server supports, it just
 changes PySTAC client's understanding of what the server supports.
 
+Configuring retry behavior
+--------------------------
+
+By default, **pystac-client** will retry requests that fail DNS lookup or have timeouts.
+If you'd like to configure this behavior, e.g. to retry on some ``50x`` responses, you can configure the StacApiIO's session:
+
+.. code-block:: python
+
+    from requests.adapters import HTTPAdapter
+    from urllib3 import Retry
+
+    from pystac_client import Client
+    from pystac_client.stac_api_io import StacApiIO
+
+    retry = Retry(total=5, backoff_factor=1, status_forcelist=[502, 503, 504])
+    stac_api_io = StacApiIO()
+    stac_api_io.session.mount("https://", HTTPAdapter(max_retries=retry))
+    client = Client.open(
+        "https://planetarycomputer.microsoft.com/api/stac/v1", stac_io=stac_api_io
+    )
+
+Automatically modifying results
+-------------------------------
+
+Some systems, like the `Microsoft Planetary Computer <http://planetarycomputer.microsoft.com/>`__,
+have public STAC metadata but require some `authentication <https://planetarycomputer.microsoft.com/docs/concepts/sas/>`__
+to access the actual assets.
+
+``pystac-client`` provides a ``modifier`` keyword that can automatically
+modify the STAC objects returned by the STAC API.
+
+.. code-block:: python
+
+   >>> from pystac_client import Client
+   >>> import planetary_computer, requests
+   >>> catalog = Client.open(
+   ...    'https://planetarycomputer.microsoft.com/api/stac/v1',
+   ...    modifier=planetary_computer.sign_inplace,
+   ... )
+   >>> item = next(catalog.get_collection("sentinel-2-l2a").get_all_items())
+   >>> requests.head(item.assets["B02"].href).status_code
+   200
+
+Without the modifier, we would have received a 404 error because the asset
+is in a private storage container.
+
+``pystac-client`` expects that the ``modifier`` callable modifies the result
```

Comment:
Would it make sense to sketch out a custom modifier?

Reply:
Makes sense, while we're touching these lines. I'll add.

Reply:
Doc update blocked by https://github.com/stac-utils/pystac-client/issues/533